TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as

```diff
@@ -87,118 +87,133 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
+		bool isBullet = (
+			customData == HittersTC::bullet_low_cal || customData == HittersTC::bullet_high_cal || 
+			customData == HittersTC::shotgun || customData == HittersTC::railgun_lance);
+		
+		string headname = this.get_string("equipment_head");
+		string torsoname = this.get_string("equipment_torso");
+		string bootsname = this.get_string("equipment_boots");
+		
+		if (headname != "" && this.exists(headname+"_health"))
 		{
-			f32 armorMaxHealth = 25.0f;
-			f32 armorHealth = armorMaxHealth - this.get_f32("bpv_health");
-			f32 ratio = armorHealth / armorMaxHealth;
+			f32 armorMaxHealth;
+			f32 ratio;
 
-			switch (customData)
+			if (headname == "militaryhelmet" && customData != HittersTC::radiation)
 			{
-				case HittersTC::bullet_low_cal:
-					ratio *= 0.80f;
-					break;
-
-				case HittersTC::bullet_high_cal:
-				case HittersTC::railgun_lance:
-					ratio *= 0.60f;
-					break;
-
-				case HittersTC::shotgun:
-					ratio *= 0.85f;
-					break;
-
-				default:
-					ratio *= 0.40f;
-					break;
+				armorMaxHealth = 40.0f;
+				switch (customData)
+				{
+					case HittersTC::bullet_low_cal:
+					case HittersTC::shotgun:
+						ratio = 0.85f;
+						break;
+
+					case HittersTC::bullet_high_cal:
+					case HittersTC::railgun_lance:
+						ratio = 0.65f;
+						break;
+
+					default:
+						ratio = 0.30f;
+						break;
+				}
 			}
-			
-			this.set_f32("bpv_health", this.get_f32("bpv_health") + (ratio*dmg));
-			f32 playerDamage = Maths::Clamp((1.00f - ratio) * dmg, 0, dmg);
-
-			dmg = playerDamage;
-		}
-		else if (this.get_string("equipment_torso") == "keg" && !isBullet && customData != HittersTC::radiation)
-		{
-			f32 armorMaxHealth = 7.0f;
-			f32 armorHealth = armorMaxHealth - this.get_f32("keg_health");
-			f32 ratio = armorHealth / armorMaxHealth;
-
-			if ((customData == Hitters::fire || customData == Hitters::burn || customData == Hitters::explosion || 
-				customData == Hitters::bomb || customData == Hitters::bomb_arrow) && this.get_f32("keg_explode") == 0.0f)
+			else if (headname == "scubagear")
 			{
-				this.set_f32("keg_explode", getGameTime() + (30.0f * 1.0f));
-				this.SetLightRadius(this.get_f32("explosive_radius") * 0.5f);
-				this.getSprite().PlaySound("/Sparkle.ogg", 1.00f, 1.00f);
-				this.getSprite().PlaySound("MigrantScream1.ogg", 1.00f, this.getSexNum() == 0 ? 1.0f : 2.0f);
-				ratio *= 0.0f;
+				armorMaxHealth = 10.0f;
+				ratio = 0.20f;
 			}
-			else ratio *= 0.50f;
-			
-			this.set_f32("keg_health", this.get_f32("keg_health") + (ratio*dmg));
-			f32 playerDamage = Maths::Clamp((1.00f - ratio) * dmg, 0, dmg);
+			else if (!isBullet && customData != HittersTC::radiation)
+			{
+				if (headname == "bucket") armorMaxHealth = 10.0f;
+				else if (headname == "pumpkin") armorMaxHealth = 5.0f;
+				else if (headname == "minershelmet") armorMaxHealth = 10.0f;
+				ratio = 0.20f;
+			}
+			f32 armorHealth = armorMaxHealth - this.get_f32(headname+"_health");
```

Comment:
What if headname+_health does not exist, then it cause a divide by 0 error

Reply:
![image](https://user-images.githubusercontent.com/68712994/126392745-a03329c5-7e21-4c36-99e4-5191f3e9ab0e.png)